src/lib/statistics/table_statistics.cpp

```diff
@@ -72,13 +114,22 @@ std::shared_ptr<TableStatistics> TableStatistics::from_table(const Table& table)
     thread.join();
   }
 
-  return std::make_shared<TableStatistics>(std::move(column_statistics), table.row_count());
+  return std::make_shared<TableStatistics>(std::move(column_statistics), std::move(segment_statistics),
```

Comment:
I don't have an opinion yet, but for my taste keeping the segment_statistics around here in the statistics objects smells. They are totally irrelevant for all users of the statistics object.

Reply:
We also considered storing the segment histograms somewhere else, but having them in the statistics object seemed to make the most sense. It'd also allow to let other users of the statistics use these, if available.

Reply:
Please don't stop discussion by resolving preemptively.